packages/vm/src/evm/evm.ts

```diff
@@ -408,7 +408,16 @@ export default class EVM {
         this._vm._common.isActivatedEIP(3541) &&
         result.returnValue.slice(0, 1).equals(Buffer.from('EF', 'hex'))
       ) {
-        result = { ...result, ...INVALID_BYTECODE_RESULT(message.gasLimit) }
+        if (!this._vm._common.isActivatedEIP(3540)) {
+          result = { ...result, ...INVALID_BYTECODE_RESULT(message.gasLimit) }
+        }
+        // EIP-3540 EOF1 checks
+        if (!eof1CodeAnalysis(result.returnValue)) {
```

Comment:
I don't think I like this function signature, so this code path will run if the `eof1CodeAnalysis` returns `undefined`, but I'm also not super sure what would make this more clean... 🤔